### PR TITLE
Add usable validation errors

### PR DESCRIPTION
**Pull request by Deconstrained**
If a line is invalid, it won't throw an error when attempting to transpose but rather will print out the line number and content that is invalid.

### CODE REVIEW

File: lib/code_owners.rb [hunk 1]
```diff
@@ -35,11 +38,25 @@ def search_codeowners_file
...
-        line.gsub(/#.*/, '').gsub(/^$/, " @").split(/\s+@/, 2)
-      end
+      i = 0
+      patterns = []
+      File.read(codeowner_path).split("\n").each { |line|
+        i += 1
+        path_owner = line.split(/\s+@/, 2)
+        if line.match(/^\s*#.*$/) || line.match(/^\s*$/)
+          patterns.push ['', ''] # Comment/empty line
+        elsif path_owner.length != 2 || path_owner[0].empty? && !path_owner[1].empty?
```
Comment (jcheatham) [inline lib/code_owners.rb]: I recommend wrapping your intended operator precedence with parens, helps with legibility, people not intimately familiar with Ruby's precedence, etc.  i.e. should it be `(A || B) && C` or `A || (B && C)`?

File: lib/code_owners.rb [hunk 2]
```diff
@@ -35,11 +38,25 @@ def search_codeowners_file
...
     # read the github file and spit out a slightly formatted list of patterns and their owners
+    # Empty/invalid/commented lines are still included in order to preserve line numbering
     def pattern_owners
       codeowner_path = search_codeowners_file
-      File.read(codeowner_path).split("\n").map do |line|
-        line.gsub(/#.*/, '').gsub(/^$/, " @").split(/\s+@/, 2)
-      end
+      i = 0
+      patterns = []
+      File.read(codeowner_path).split("\n").each { |line|
```
Comment (jcheatham) [inline lib/code_owners.rb]: can replace `each` with `each_with_index` and drop the i setting and updater

File: lib/code_owners.rb [hunk 3]
```diff
@@ -35,11 +38,25 @@ def search_codeowners_file
...
+        i += 1
+        path_owner = line.split(/\s+@/, 2)
+        if line.match(/^\s*#.*$/) || line.match(/^\s*$/)
```
Comment (jcheatham) [inline lib/code_owners.rb]: would likely be faster (better?) to have the OR happen in the regex, e.g. `line.match(/^\s*(?:#.*)?$/`

File: lib/code_owners.rb [hunk 4]
```diff
@@ -13,14 +13,17 @@ class << self
...
 
-    def ownerships
-      patterns, owners = pattern_owners.transpose
+    def log(message)
+      warn message
+    end
 
-      git_owner_info(patterns).map do |line, pattern, file|
+    def ownerships
+      patterns = pattern_owners
```
Comment (jcheatham) [inline lib/code_owners.rb]: admittedly `.transpose` was not very clear, but variable name here is a little misleading... maybe rename the method to `parse_pattern_owners` or `get_pattern_owners`?

File: lib/code_owners.rb (hunk 2 above)
Comment (Deconstrained) [reply lib/code_owners.rb]: Ah thanks! I was searching for something similar to Python's `enumerate` but was failing to grasp at the correct terms / find the right documentation.

**Pull request merged**